src/main/java/com/bobocode/controller/UrlShortenerController.java

```diff
@@ -0,0 +1,43 @@
+package com.bobocode.controller;
+
+
+import static org.springframework.http.HttpStatus.MOVED_PERMANENTLY;
+
+import com.bobocode.dto.UrlInfoRequest;
+import com.bobocode.service.UrlShortenerService;
+import java.net.URI;
+import javax.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+@RestController
+@RequestMapping("/v1/short-url")
+@RequiredArgsConstructor
+public class UrlShortenerController {
+
+  private final UrlShortenerService urlShortenerService;
+
+  @PostMapping
+  public ResponseEntity<Object> shortenUrl(@RequestBody @Valid UrlInfoRequest urlInfo) {
+    String shortUrlKey = urlShortenerService.shorten(urlInfo);
+    return ResponseEntity.created(buildShortURI(shortUrlKey)).build();
```

Comment:
@kebann a POST request that creates something should probably return status `201`

Reply:
`created` sets the status to `201` under the hood